### ISSUE TITLE
useIsAuthenticated hook doesn't cause re-render

### ISSUE THREAD

**Issue by zarnocha** [react-auth-kit/react-auth-kit#1541]
Hello.
As in the title the useIsAuthenticated hook won't cause component's re-render.

**Comment by wiktor-maruszewski** [comment]
I've got the same problem. Please fix it quickly! :(

**Comment by darkmatter18** [comment]
Hey can you please elaborate what your intended behaviour is? I am sorry, but I can't understand

**Comment by imjulianeral** [comment]
I think I can explain. At least for my use case, when the auth state changes, my router redirects automatically without the need of an explicit redirect, why? because with `useIsAuthenticated`  hook  I can conditionally render what I need if the user is signed in or not, but since v3 this hook which is expected to trigger a rerender because it is a state handler which changes if the user is auth or not the automatic redirect I got working with v2 is not working with v3.

So basically the hook isn't getting the last value automatically, you need to refresh the page to get the latest auth state.

**Comment by darkmatter18** [comment]
@imjulianeral Let's have an in-person call. 
Please setup a call best on your time 
https://calendly.com/arkadip/react-auth-kit

**Comment by slugiscool99** [comment]
Same problem here - can help if needed

**Comment by slugiscool99** [comment]
Same problem here - can help if needed

**Comment by imjulianeral** [comment]
Sure thing!

**Comment by imjulianeral** [comment]
This is a repo which reproduces this issue. 

https://github.com/imjulianeral/react-auth-kit-state-bug

**Comment by jaroslavhuss** [comment]
Any update on this? @darkmatter18? Using `` "@auth-kit/react-router": "^3.0.2-alpha.19" `` none of the hooks singIn/signOut causes rerender of `` useIsAuthenticated() ``. As @imjulianeral mentioned - windows muset be reloaded in order to get the state.

**Comment by ThaerHindawi** [comment]
I have the same problem.
Is there any new update?

**Comment by kennethhaspeel** [comment]
I don't anything will change. It seems the current version doesn't use state so there is no state update. 

I draw this conclusion from this: [https://github.com/react-auth-kit/react-auth-kit/issues/1558](url)

**Comment by imjulianeral** [comment]
@kennethhaspeel Wow, that's not what the mantainer said to me on our meet. Maybe the right move is to stick with v2.

**Comment by mahelbir** [comment]
I have the same problem.
Is there any new update?

**Comment by darkmatter18** [comment]
The fix is coded on the master branch. Please wait for few more days for upcoming release

**Comment by kennethhaspeel** [comment]
> Is there a temporary workaround we can use before V4 release?

Have you tried the workaround @Odinaev444 mentioned? I'm not using this in my current project anymore (but will use it in future projects) so not sure if it works for your problem but I remember it working for me two months ago

**Comment by kvandake** [comment]
I also have a similar problem...

**Comment by nikolay-emrikh-stenn** [comment]
Will the fix of this issue be released any day?

**Comment by CinDevFC** [comment]
i have the same problem, has anybody fixed it?

**Comment by tomcarter17** [comment]
@darkmatter18 is there any intent to release v4 soon? If not, would it be possible to release this fix in a minor version? The current bug seems to break the way the hook is intended to be used based upon the [example](https://authkit.arkadip.dev/reference/react-auth-kit/hooks/useIsAuthenticated/) provided in the documentation.

**Comment by JustinKubicz** [comment]
Any updates? As of 12/6 this is still not working.

**Comment by RedCore161** [comment]
my project is broken now. probablly will switch libary. i really liked this one, but thats a deal breaker
